byway/src/main/java/com/google/sps/servlets/LoginServlet.java

```diff
@@ -0,0 +1,64 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import java.io.IOException;
+import java.util.ArrayList;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import com.google.sps.data.Login;
+import com.google.sps.data.Trip;
+
+
+@WebServlet("/login")
+public class LoginServlet extends HttpServlet {
+  private final Gson gson = new Gson();
+  private final UserService userService = UserServiceFactory.getUserService();
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+  private static final String URL_TO_REDIRECT_TO_AFTER_LOGSIN = "/destinations.html";
+  private static final String URL_TO_REDIRECT_TO_AFTER_LOGSOUT = "/index.html";
+
+  private Key tripKey;
+    
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    createTripEntity();
```

Comment:
The login page creates new trip entities? So overtime a user logs into the website, they automatically create a brand new trip? This seems like a bad design. 

The user should have to explicitly create trip in some way. Automatically creating trips for them every time they open the page will inevitably lead to undesired behavior and side affects. See my recommendation to create new Trips via the destinations page servlet.

Reply:
fixed